### DISCUSSION TITLE
ChatGPT-like input

### DISCUSSION THREAD

**Discussion by dominikpeter** [zauberzeug/nicegui#3704] [Q&A] [answered]
### Question

I am looking for a way to mimic the behaviour of ChatGPT Input - or most other chatbots. This behaves as follows: When typing text and pressing Enter, the "search" is fired. When hitting Shift+Enter or pasting line delimited text, the input automatically becomes multiline.

I can easily enable autogrow with quasar props. However, this now also adds a new line with a normal `keypress.enter`. 

```python
ui.input(placeholder="Search").props(
    """
    rounded 
    autogrow
    autofocus 
    outlined
    v-model="search" 
    type="search" 
    """
 )
```
I also found this on [Stackoverflow](https://stackoverflow.com/questions/53571707/how-to-handle-an-enter-in-a-q-input-preventdefault-on-submit) but could not get it to work.

Any idea how I can change the props to enable this behaviour?

**Comment by python-and-novella** [comment, marked as answer]
```python3
ui.input(placeholder="Search").props(
    """
    rounded 
    autogrow
    autofocus 
    outlined
    """
 ).on('keypress.enter',lambda e:None if e.args['shiftKey'] else e.sender.set_value('')  ,args=['shiftKey'])
```

**Comment by dominikpeter** [comment]
Awesome! Thanks a lot!

**Reply by dominikpeter** [reply]
The code works. But the input now has these glitches that are annoying. When you press Enter, it wants to jump to the new line, but then makes the call directly. Does anyone know how to avoid this?

![search](https://github.com/user-attachments/assets/f7707114-3b5c-4f2e-babf-b7bb92a8af73)

**Reply by python-and-novella** [reply]
Use javascript's event.preventDefault().
But I have no time to show example.

**Reply by aersam** [reply]
I have time ;)

preventDefault only seems to work with a js_handler, so you need two handlers:

```python
.on("keypress", self._handle_keypress) \
.on("keypress", js_handler="""
    (e) => { 
        if (!e.shiftKey && e.key === "Enter") { 
            e.preventDefault()
        }
    }
""")
```

**Reply by dominikpeter** [reply]
This works :-)
